python-connectors/kayrros_retrieve-dataset/connector.py

```diff
@@ -0,0 +1,123 @@
+# This file is the actual code for the custom Python dataset kayrros_retrieve-dataset
+
+from dataiku.connector import Connector
+from utils.authentification import get_headers
+import requests
+import dataiku
+
+class MyConnector(Connector):
+    
+    # A custom Python dataset is a subclass of connector.
+
+    def __init__(self, config, plugin_config):
+        
+        """
+        The configuration parameters set up by the user in the settings tab of the
+        dataset are passed as a json object 'config' to the constructor.
+        The static configuration parameters set up by the developer in the optional
+        file settings.json at the root of the plugin directory are passed as a json
+        object 'plugin_config' to the constructor
+        """
+        
+        Connector.__init__(self, config, plugin_config) 
+
+        # Perform some more initialization
+        self.id = self.config["id_dataset"]
+        self.preset = self.config["preset"]
```

Comment:
Add checks on the inputs.

```suggestion
        self.id = self.config["id_dataset"]
        if self.id == "":
            raise ValueError("A Kayross dataset ID is necessary to fetch the data. Please provide one in the plugin settings.")
       
        self.preset = self.config["preset"]
        if not self.preset:
            raise ValueError("A Kayrros account is necessary to fetch the data. Please provide one in the preset field.")
```